### PR TITLE
backendutil: Implement message size and lines counting

### PR DESCRIPTION
**Pull request by foxcpp**

### CODE REVIEW

File: backend/backendutil/bodystructure.go [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
 	"mime"
 	"strings"
 
 	"github.com/emersion/go-imap"
 	"github.com/emersion/go-message/textproto"
 )
 
+type countReader struct {
+	r           io.Reader
+	total       uint32
```
Comment (emersion) [inline backend/backendutil/bodystructure.go:17]: Maybe consider renaming this to `bytes`, to make it clear what the count is.

File: backend/backendutil/bodystructure.go [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
+	newlines    uint32
+	lastNewline bool
+}
+
+func (r *countReader) Read(b []byte) (int, error) {
+	n, err := r.r.Read(b)
+	r.total += uint32(n)
+	if n != 0 {
+		r.newlines += uint32(bytes.Count(b[:n], []byte{'\r'}))
```
Comment (emersion) [inline backend/backendutil/bodystructure.go:26]: Any reason why we count `\r` instead of `\n` here?

File: backend/backendutil/bodystructure.go [hunk 3]
```diff
@@ -48,14 +73,41 @@ func FetchBodyStructure(header textproto.Header, body io.Reader, extended bool)
...
+			bufBody := bufio.NewReader(&countedBody)
+			subMsgHdr, err := textproto.ReadHeader(bufBody)
+			if err != nil {
+				return nil, err
+			}
+			bs.Envelope, err = FetchEnvelope(subMsgHdr)
+			if err != nil {
+				return nil, err
+			}
+			// TODO: Limit recursion.
```
Comment (emersion) [inline backend/backendutil/bodystructure.go:91]: I wonder if we should do this in go-message directly. Would probably be a good idea.

File: backend/backendutil/bodystructure.go (hunk 2 above)
Comment (foxcpp) [reply backend/backendutil/bodystructure.go:26]: Woops, right. It is supposed to be `\n` but I forgot to change it back after checking something.

File: backend/backendutil/bodystructure.go (hunk 3 above)
Comment (foxcpp) [reply backend/backendutil/bodystructure.go:91]: I will drop that TODO, I think we have an issue open in go-message already.

**Pull request merged**